test/test_StateMachine.py

```diff
@@ -367,3 +367,34 @@ def test_reset(self):
         self.assertEqual(sm.state, 'bar')
         sm.reset()
         self.assertIsNone(sm.state)
+
+    def test_can(self):
+        sm = StateMachine({
+            'initial': 'init',
+            'transitions': {
+                ('init', 'foo'): lambda: True,
+                ('foo', 'bar'): lambda: True,
+                ('bar', 'foo'): lambda: True,
+                ('bar', 'init'): lambda: True
+            }
+        })
+
+        self.assertEqual(sm.state, None)
+        self.assertIs(sm.can('init'), True)
+        self.assertIs(sm.can('foo'), False)
+        self.assertIs(sm.can('bar'), False)
+
+        sm.process(0)
+        self.assertEqual(sm.state, 'init')
+        self.assertIs(sm.can('foo'), True)
+        self.assertIs(sm.can('bar'), False)
```

Comment:
What about the case when you're in `'init'` and you call `can('init')` (and the same for other states)?

That should always return `False`, right? Theoretically, I think the StateMachine does support a transition that loops back on itself... though I can't really think of a scenario where it would be useful. Every cycle calls `in_state` anyway, but that would force an `on_exit` and `on_entry` when the transition is triggered.

Anyway, do you think it's worthwhile testing for `self.assertIs(sm.can(sm.state), False)` here?